memory/ndctl.py

```diff
@@ -279,33 +285,67 @@ def test_multiple_namespaces_region(self):
         """
         Test multiple namespace with single region
         """
-        max_cnt = cnt = self.params.get('namespace_cnt', default=5)
+        max_cnt = self.cnt
         self.enable_region()
         region = self.get_json(short_opt='-R')[0]
         self.log.info("Using %s for muliple namespace regions", region)
         self.disable_namespace(region=region)
         self.destroy_namespace(region=region)
         size = self.get_json_val(self.get_json(
             long_opt='-r %s' % region)[0], 'size')
-        self.log.info("Creating %s namespaces", cnt)
+        self.log.info("Creating %s namespaces", self.cnt)
         if 'Hash' in genio.read_file('/proc/cpuinfo').rstrip('\t\r\n\0'):
             def_align = 16 * 1024 * 1024
         else:
             def_align = 2 * 1024 * 1024
-        if ((size / cnt) % def_align) != 0:
+        if ((size / self.cnt) % def_align) != 0:
             self.log.warn("Namespace would fail as it is not %sM "
                           "aligned! Chaning the number of "
                           "namespaces", def_align / (1024 * 1024))
             for count in range(max_cnt, 1, -1):
                 if ((size / count) % def_align) == 0:
                     self.log.info("Changing namespaces to %s", count)
-                    cnt = count
+                    self.cnt = count
                     break
 
-        for nid in range(0, cnt):
-            self.create_namespace(region=region, mode='fsdax', size=size / cnt)
+        for nid in range(0, self.cnt):
+            self.create_namespace(
+                region=region, mode='fsdax', size=size / self.cnt)
             self.log.info("Namespace %s created", nid + 1)
 
+    def test_multiple_ns_multiple_region(self):
+        """
+        Test multiple namespace with multiple region
+        """
+        max_cnt = self.cnt
+        self.enable_region()
+        regions = self.get_json(short_opt='-R')
+        self.disable_namespace()
+        self.destroy_namespace()
+        for region in regions:
+            self.log.info("Using %s for muliple namespaces", region)
+            size = self.get_json_val(self.get_json(
+                long_opt='-r %s' % region)[0], 'size')
+            self.log.info("Creating %s namespaces", self.cnt)
+            if 'Hash' in genio.read_file('/proc/cpuinfo').rstrip('\t\r\n\0'):
```

Comment:
This check is being done twice in this script. Better to move it into a helper function.